compiler/esy.json

```diff
@@ -4,13 +4,15 @@
   "esy": {
     "buildEnv": {
       "DUNE_BUILD_DIR": "#{self.target_dir}",
-      "GRAIN_STDLIB": "#{self.root}/../stdlib"
+      "GRAIN_STDLIB": "#{self.root / '..' / 'stdlib'}",
+      "GRAINC_BIN_PATH": "#{self.target_dir / 'default' / 'grainc' / 'grainc.exe'}",
+      "CLI_BIN_DIR": "#{self.root / '..' / 'cli' / 'bin'}"
```

Comment:
I switched this stuff to esy's path expansion to hopefully help with Windows support of the commands.